lib/root_controller.js

```diff
@@ -0,0 +1,174 @@
+var Tessel = require('./tessel/tessel');
+var logs = require('./logs');
+var debug = require('debug')('controller');
+
+// create an object for t2 root steps for beeing more handy about unit testing of the T2 root command itself.
+var controller = {};
+controller.ssh = {
+  makeMenu: function(tessel, index, rtm) {
+    if (tessel.connection.connectionType === 'LAN') {
+      if (!tessel.connection.authorized) {
+        rtm.add(index + ') ' + tessel.name + ': ' + tessel.connection.ip + ' (not authorized) \n');
+      } else {
+        rtm.add(index + ') ' + tessel.name + ': ' + tessel.connection.ip + ' (authorized) \n');
+      }
+
+    } else if (tessel.connection.connectionType === 'USB') {
+      if (!tessel.connection.authorized) {
+        rtm.add(index + ') ' + tessel.name + ': [USB] (not authorized) \n');
+      } else {
+        rtm.add(index + ') ' + tessel.name + ': [USB] (authorized) \n');
+      }
+    } else {
+      rtm.add('EXIT');
+    }
+    // FIXME: Workaround for trouble with dynamic added menu elements
+    rtm._draw();
+  },
+  runSSH: function(id, opts, tessels, resolve, reject) {
+    // clear console
+    debug('runSSH...\n root@' + tessels[id].connections[0].host);
+    if (opts.menu) {
+      // because the test needs to resolve the promise ...
+      resolve({
+        opts: opts,
+        tessels: tessels
+      });
+    } else {
+      // if this is no test, starting the menu as child process
+      process.stdout.write('\u001b[2J\u001b[0;0H');
+      var ch = require('child_process')
+        .spawn('ssh', ['-i',
+          opts.path,
+          'root@' + tessels[id].connections[0].ip
+        ], {
+          stdio: 'inherit'
+        });
+      logs.info('Connect to ' + tessels[id].connections[0].host + '...');
+
+      // FIXME: There is no handler for poweroff Tessel ! Needs to be fixed within the firmware...
+      // (Terminal freezes imediatelly after poweroff the Tessel)
+
+      ch.once('error', function(e) {
+        if (e === 255) {
+          process.stdout.write('\u001b[2J\u001b[0;0H');
+          logs.warn('Sorry, you are not authorized!');
+          logs.info('"t2 key generate" might help :-)');
```

Comment:
@Student007 if the code gets to this point, it should have already run through `Tessel.get` which weeds out Tessels that aren't authorized. This case should never be hit.